File: backend/LeaveManagerAPI/Models/LeaveRequest.cs
namespace LeaveManagerAPI.Models
{
    public enum LeaveRequestStatus
    {
        Pending,
        Approved,
        Rejected,
        Cancelled
    }
    public class LeaveRequest
    {
        public int Id { get; set; }

        public string UserId { get; set; } = string.Empty;

        public User User { get; set; }

        public LeaveType Type { get; set; }

        public DateOnly StartDate { get; set; }

        public DateOnly EndDate { get; set; }

        public int RequestedDays { get; set; }

        public string? Reason { get; set; }

        public LeaveRequestStatus Status { get; set; }

        public string? ReviewerId { get; set; }

        public User Reviewer { get; set; }

        public DateTime? ReviewedAt { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: frontend/src/app/models/leave-request.ts
export interface CreateLeaveRequestDto{
    type: number,
    startDate: string,
    endDate: string,
    reason: string | null
}

export interface LeaveRequestDto{
    id: number,
    type: string,
    startDate: string,
    endDate: string,
    requestedDays: number,
    reason: string | null,
    status: string,
    reviewerName: string | null,
    reviewedAt: string | null,
    createdAt: string,
    requesterName: string
}

export interface FilterLeaveRequestCalendar {
    startDate: string,
    endDate: string
}

export interface LeaveRequestCalendar{
    id: number,
    type: string,
    startDate: string,
    endDate: string,
    status: string,
    requesterName: string
}